examples/async_tasks/async_compute.rs

```diff
@@ -24,6 +24,8 @@ const NUM_CUBES: u32 = 6;
 
 struct BoxMeshHandle(Handle<Mesh>);
 struct BoxMaterialHandle(Handle<StandardMaterial>);
+#[derive(Component)]
+struct FinalPositionTask(Task<Transform>);
```

Comment:
Why is it necessary to wrap `Task` in a newtype? `.insert()` did expect a `Component`, so shouldn't `Task` just work?

Reply:
`Task` should not be a `Component`; indeed that impl is removed in this PR (due to an old version of this PR making the impl be done by the derive macro, it wouldn't have worked for the impl).

See also #2982

Perhaps this change should be moved to #2982, but it's not too important.

Reply:
Ok, thanks for the explanation.